region/brinstar/red/Red Brinstar Fireflea Room.json

```diff
@@ -1143,10 +1160,75 @@
       "link": [2, 2],
       "name": "Crystal Flash",
       "requires": [
+        {"obstaclesNotCleared": ["R-Mode"]},
         "h_CrystalFlash"
       ],
       "flashSuitChecked": true
     },
+    {
+      "link": [2, 2],
+      "name": "R-Mode Spark Interrupt",
+      "requires": [
+        {"obstaclesCleared": ["R-Mode"]},
+        {"or": [
+          "h_CrystalFlashForReserveEnergy",
+          {"and": [
+            "h_RModeCanRefillReserves",
+            {"resourceMissingAtMost": [{"type": "PowerBomb", "count": 0}]},
+            {"partialRefill": {"type": "ReserveEnergy", "limit": 40}}
+          ]}
+        ]},
+        {"canShineCharge": {"usedTiles": 13, "openEnd": 0}},
+        {"autoReserveTrigger": {}},
+        "canRModeSparkInterrupt"
+      ],
+      "resetsObstacles": ["R-Mode"],
+      "flashSuitChecked": true,
+      "blueSuitChecked": true,
+      "note": [
+        "Farm no more than one fireflea: Samus sprite and spike tiles are darkened by the Fireflea effect during R-Mode.",
+        "Shinecharge in the right pit next to the spikes, then jump up to the fireflea to interrupt."
+      ]
+    },
+    {
+      "link": [2, 2],
+      "name": "X-Mode Shinecharge, R-Mode Spark Interrupt",
+      "entranceCondition": {
+        "comeInWithRMode": {}
+      },
+      "requires": [
+        {"obstaclesCleared": ["R-Mode"]},
+        {"or": [
+          "h_CrystalFlashForReserveEnergy",
+          {"and": [
+            "h_RModeCanRefillReserves",
+            {"resourceMissingAtMost": [{"type": "PowerBomb", "count": 0}]},
+            {"partialRefill": {"type": "ReserveEnergy", "limit": 40}}
+          ]}
+        ]},
+        "h_spikeXModeShinecharge",
```

Comment:
Its canBeVeryPatient because of the xmode.  But it might also be patient for damaging down on wavers.